### PR TITLE
Added separate methods for post and get for user intent

### PR DESCRIPTION
**Pull request by jyasveer**

### CODE REVIEW

**Comment by msrb** [comment]
unit tests are failing: https://ci.centos.org/job/devtools-fabric8-analytics-server-fabric8-analytics/86/console

File: bayesian/api_v1.py [hunk 1]
```diff
@@ -679,6 +692,7 @@ def post():
 add_resource_no_matter_slashes(StackAnalysesGET, '/stack-analyses/<external_request_id>')
 add_resource_no_matter_slashes(UserFeedback, '/user-feedback')
 add_resource_no_matter_slashes(UserIntent, '/user-intent')
+add_resource_no_matter_slashes(UserIntentGet, '/user-intent/<user>/<ecosystem>')
```
Comment (fridex) [inline bayesian/api_v1.py]: What about adjusting only the line above? That should do the trick. Everything else can stay untouched.

Comment (jyasveer) [reply bayesian/api_v1.py]: how do you intent to do it ?

File: bayesian/api_v1.py [hunk 2]
```diff
@@ -425,26 +425,11 @@ class UserIntent(ResourceWithSchema):
...
-                                                                                    e=ecosystem)
-            current_app.logger.exception(err_msg)
-            raise HTTPError(404, error=err_msg)
-
-        return result
+        raise HTTPError(404, "Unsupported API endpoint")
 
     @staticmethod
     def post():
+        decoded = decode_token()
```
Comment (msrb) [inline bayesian/api_v1.py]: Do we need this here?

File: bayesian/api_v1.py [hunk 3]
```diff
@@ -425,26 +425,11 @@ class UserIntent(ResourceWithSchema):
...
-        try:
-            result = s3.fetch_user_data(user, ecosystem)
-        except botocore.exceptions.ClientError:
-            err_msg = "Failed to fetch data for the user {u}, ecosystem {e}".format(u=user,
-                                                                                    e=ecosystem)
-            current_app.logger.exception(err_msg)
-            raise HTTPError(404, error=err_msg)
-
-        return result
+        raise HTTPError(404, "Unsupported API endpoint")
```
Comment (msrb) [inline bayesian/api_v1.py]: 405 would be probably more appropriate here. But this should be transparently handled by flask (webserver? not sure). So we can remove the `get()` method altogether.

File: bayesian/api_v1.py [hunk 4]
```diff
@@ -474,6 +459,34 @@ def post():
...
+            err_msg = "Failed to fetch data for the user {u}, ecosystem {e}".format(u=user,
+                                                                                    e=ecosystem)
+            current_app.logger.exception(err_msg)
+            raise HTTPError(404, error=err_msg)
+
+        return result
+
+    @staticmethod
+    def post():
+        raise HTTPError(404, "Unsupported API endpoint")
```
Comment (msrb) [inline bayesian/api_v1.py]: 405, or remove :) I'd prefer removing this method.

File: bayesian/api_v1.py (hunk 1 above)
Comment (fridex) [reply bayesian/api_v1.py]: See https://github.com/fabric8-analytics/fabric8-analytics-server/pull/186 if that would work for you.

Comment (jyasveer) [reply bayesian/api_v1.py]: if i have a single end point , how will my post request get served ? @fridex

Comment (fridex) [reply bayesian/api_v1.py]: Agh, good point, I missed the POST endpoint. PR has been adjusted, please take a look.

**Pull request merged**